src/packages/database/model/index.js

```diff
@@ -592,6 +592,16 @@ class Model {
   static primaryKey: string = 'id';
 
   /**
+   * allows the internal hooks to define the id value
+   *
+   * @property allowPrimaryKeyInsert
+   * @type {boolean}
+   * @default 'id'
+   * @public
+   */
+  allowPrimaryKeyInsert: boolean = false;
```

Comment:
I think that this flag is unnecessary. The ORM should be smart enough to know if a record has an id in it's change set and it has not been persisted to the database yet.